File: DynamicProgramming/22houseRobber/houseRobber.java

class houseRobber {
    public static int rob(int[] nums) {
        int r1 = 0,
                r2 = 0,
                temp = 0,
                Len = nums.length;
        for (int i = 0; i < Len; i++) {
            r1 = nums[i] + r1;
            temp = r1 > r2 ? r1 : r2;
            r1 = r2;
            r2 = temp;
        }
        return r2;
    }

    public static void main(String[] args) {
        int[] nums = { 2, 7, 9, 3, 1 };
        System.out.println(rob(nums));
    }
}

File: DynamicProgramming/22houseRobber/houseRobber.py
# House Robber (leetcode 198) 
class Solution:
    def rob(self, nums: list[int]) -> int:
        rob1, rob2 = 0, 0

        # [rob1 , rob2, n, n+1, ....]
        for n in nums:
            temp = max(n + rob1, rob2)
            rob1 = rob2
            rob2 = temp
        return rob2
print(Solution().rob([1,2,3,1]))
print(Solution().rob([1,6,1,1,12,1]))
print(Solution().rob([1,2,3]))
print(Solution().rob([2,3,2]))
print(Solution().rob([6,1,1,1,1,12]))
